### PR TITLE
Add segment library integration with Gradebook to track events

### PR DESCRIPTION
**Pull request by schenedx**
@edx/educator-neem Please review.
This PR mimic what's done in https://github.com/edx/edx-portal/pull/29.
It's not meant to be a complete capture of all the events on gradebook. This is an integration to setup tracking on gradebook

All the events should stream into segment.io after deployment.

### CODE REVIEW

File: src/segment.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    return function () {
+      const args = Array.prototype.slice.call(arguments);
+      args.unshift(method);
+      analytics.push(args);
+      return analytics;
+    };
+  };
+
+  // For each of our methods, generate a queueing stub.
+  for (let i = 0; i < analytics.methods.length; i++) {
```
Comment (iloveagent57) [inline src/segment.js]: Isn't this just `analytics.methods.forEach(methodName => analytics[methodName] = analytics.factory(methodName);)`?

File: src/segment.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  // Define a factory to create stubs. These are placeholders
+  // for methods in Analytics.js so that you never have to wait
+  // for it to load to actually record data. The `method` is
+  // stored as the first argument, so we can replay the data.
+  analytics.factory = function (method) {
+    return function () {
+      const args = Array.prototype.slice.call(arguments);
+      args.unshift(method);
+      analytics.push(args);
```
Comment (iloveagent57) [inline src/segment.js]: I don't understand this.

File: src/data/store.js [hunk 3]
```diff
@@ -2,14 +2,42 @@ import { applyMiddleware, createStore } from 'redux';
...
+  [GOT_GRADES]: trackEvent(action => ({
+    name: 'Grades displayed or paginated',
+    properties: {
+      track: action.track,
+      cohort: action.cohort,
+      prev: action.prev,
+      next: action.next,
+    },
+  })),
+  [GRADE_UPDATE_SUCCESS]: trackEvent(action => ({
```
Comment (iloveagent57) [inline src/data/store.js]: Is it worth tracking the failures, too?

Comment (schenedx) [reply src/data/store.js]: Good point.

File: src/segment.js (hunk 2 above)
Comment (schenedx) [reply src/segment.js]: This is not my code. This whole file comes from segment.js. I rather not making changes here.

Comment (iloveagent57) [reply src/segment.js]: Yeah, good point.  I am curious in general about this type of pattern, but I can look it up on the internet.

File: src/data/store.js (hunk 3 above)
Comment (schenedx) [reply src/data/store.js]: @iloveagent57 This is now eventing on both success and failure cases.

**Pull request merged**